File: Logitude/AngularModules/AngularModules/Infrastructure/DataContracts/AutomationSendInterface.ts
export class AutomationSendInterface {

    constructor() {
        this.FTPDetails = new FTPAutomationDetails();
        this.WebHookDetails = new WebHookAutomationDetails();
        this.AdvancedAutomationSendInterfaceDetails = new AdvancedAutomationSendInterfaceDetails();
    }


    public InterfaceName: string;
    public SendVia: string;
    public Format: string;
    public ComputingPartnerId: string;
    public FTBFolderId: string;
    public FTPDetails: FTPAutomationDetails;
    public WebHookDetails: WebHookAutomationDetails;
    public AdvancedAutomationSendInterfaceDetails: AdvancedAutomationSendInterfaceDetails;
    public IsChanged: boolean;

}



export class FTPAutomationDetails {
    public Host: string;
    public Folder: string;
    public UserName: string;
    public Password: string;


}

export class WebHookAutomationDetails {
    public URL: string;
    public AuthenticationType: string;
    public BasicAuthUserName: string;
    public BasicAuthPassword: string;
}

export class AdvancedAutomationSendInterfaceDetails {
    public IncludeEvents: boolean;
}

File: Logitude/Simplog.Data/Helpers/AutomationSendInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Simplog.Data.Helpers
{
   
        [DataContract(Namespace = "")]
        public class AutomationSendInterface
        {

            [DataMember]
            public string InterfaceName { get; set; }

            [DataMember]
            public string SendVia { get; set; }


            [DataMember]
            public string Format { get; set; }

            [DataMember]
            public string ComputingPartnerId { get; set; }

            [DataMember]
            public FTPAutomationDetails FTPDetails { get; set; }

            [DataMember]
            public WebHookAutomationDetails WebHookDetails { get; set; }

            [DataMember]
            public AdvancedAutomationSendInterfaceDetails AdvancedAutomationSendInterfaceDetails { get; set; }
    }


        [DataContract(Namespace = "")]
        public class FTPAutomationDetails
        {
            [DataMember]
            public string Host { get; set; }
            [DataMember]
            public string Folder { get; set; }
            [DataMember]
            public string UserName { get; set; }
            [DataMember]
            public string Password { get; set; }

        }

        [DataContract(Namespace = "")]
        public class WebHookAutomationDetails
        {
            [DataMember]
            public string URL { get; set; }

        }

        [DataContract(Namespace = "")]
        public class AdvancedAutomationSendInterfaceDetails
        {
            [DataMember]
            public bool IncludeEvents { get; set; }
        }
}
